flink-runtime/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -185,6 +234,169 @@ StreamOperatorFactory<?> getHeadOperatorForNodeFromCache(StreamNode node) {
         return nodeToHeadOperatorCache.get(node);
     }
 
+    public CheckpointingMode getCheckpointingMode() {
```

Comment:
Maybe introduce a static `getCheckpointingMode(CheckpointConfig checkpointConfig)` method to be reused by `getCheckpointingMode()` and `StreamingJobGraphGenerator`.